src/main/java/dev/sigstore/fulcio/client/FulcioValidator.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2022 The Sigstore Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.sigstore.fulcio.client;
+
+import com.google.api.client.util.PemReader;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.PublicKey;
+import java.security.cert.*;
+import java.security.spec.EncodedKeySpec;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Collections;
+import java.util.Date;
+import org.conscrypt.ct.CTLogInfo;
+import org.conscrypt.ct.CertificateEntry;
+import org.conscrypt.ct.VerifiedSCT;
+
+public class FulcioValidator {
+  private final CTLogInfo ctLogInfo;
+  private final TrustAnchor fulcioRoot;
+
+  public static FulcioValidator newFulcioValidator(byte[] ctfePublicKey, byte[] fulcioRoot)
+      throws InvalidKeySpecException, NoSuchAlgorithmException, CertificateException, IOException,
+          InvalidAlgorithmParameterException {
+    KeyFactory keyFactory = KeyFactory.getInstance("EC");
```

Comment:
This actually wouldn't work for staging - Staging is using an RSA key. I don't think we should require that the CT log key type is an ECDSA key. Can this be configurable, or better yet, is there a way to easily try different key types? It will only be either an RSA, ECDSA, or EDDSA/ED25519 key.

See https://github.com/sigstore/cosign/pull/1806

Reply:
interesting, I'll add a todo and file a bug.

Reply:
filed at #4